### ISSUE TITLE
[gdal] Link error on x64-windows

### ISSUE THREAD

**Issue by cdcseacave** [Microsoft/vcpkg#5891]
Visual studio 2017 latest update
latest vcpkg: 6f0923ae93d28287d21219c57a6c8cd3e249bdd4 

release succeeds, but debug fails with unresolved external symbol of the kind `PQerrorMessage`
[nmake-build-x64-windows-debug-out.zip](https://github.com/Microsoft/vcpkg/files/3029420/nmake-build-x64-windows-debug-out.zip)

**Comment by rkcarroll** [comment]
Same issue for me as well (I think):
Linker output:
E:\Libs\vcpkg\buildtrees\gdal\src-x64-windows-debug\gdal-2.4.0>if exist *.ilk del *.ilk 
	link /nologo /dll /INCLUDE:OSRValidate  /INCLUDE:OPTGetProjectionMethods  /INCLUDE:OGR_G_GetPointCount  /INCLUDE:OGRRegisterAll /INCLUDE:GDALSimpleImageWarp  /INCLUDE:GDALReprojectImage  /INCLUDE:GDALComputeMedianCutPCT  /INCLUDE:GDALDitherRGB2PCT  /INCLUDE:OCTNewCoordinateTransformation port\*.obj gcore\*.obj alg\*.obj frmts\o\*.obj ogr\ogrsf_frmts\ogrsf_frmts.lib ogr\ogr.lib gnm\*.obj gnm\gnm_frmts\o\*.obj third_party\o\*.obj apps\commonutils.obj apps\gdalinfo_lib.obj apps\gdal_translate_lib.obj apps\gdalwarp_lib.obj apps\ogr2ogr_lib.obj  apps\gdaldem_lib.obj apps\nearblack_lib.obj apps\gdal_grid_lib.obj apps\gdal_rasterize_lib.obj apps\gdalbuildvrt_lib.obj    E:\Libs\vcpkg\installed\x64-windows\debug\lib\expat.lib                  E:\Libs\vcpkg\installed\x64-windows\debug\lib\projd.lib E:\Libs\vcpkg\installed\x64-windows\debug\lib\sqlite3.lib   E:\Libs\vcpkg\installed\x64-windows\debug\lib\geos_cd.lib        E:\Libs\vcpkg\installed\x64-windows\debug\lib\libcurl.lib wsock32.lib wldap32.lib winmm.lib    legacy_stdio_definitions.lib odbc32.lib odbccp32.lib user32.lib  E:\Libs\vcpkg\installed\x64-windows\debug\lib\libpng16d.lib   E:\Libs\vcpkg\installed\x64-windows\debug\lib\openjp2.lib          E:\Libs\vcpkg\installed\x64-windows\debug\lib\webpd.lib      E:\Libs\vcpkg\installed\x64-windows\debug\lib\libxml2.lib      ws2_32.lib  kernel32.lib psapi.lib gcore\Version.res  /out:gdal204.dll /implib:gdal_i.lib   /debug
   Creating library gdal_i.lib and object gdal_i.exp
**postgisrasterdataset.obj : error LNK2019: unresolved external symbol PQerrorMessage** referenced in function "private: int __cdecl PostGISRasterDataset::SetRasterProperties(char const *)" (?SetRasterProperties@PostGISRasterDataset@@AEAAHPEBD@Z)
postgisrasterdriver.obj : error LNK2001: unresolved external symbol PQerrorMessage
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQerrorMessage
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQerrorMessage
postgisrasterdataset.obj : error LNK2019: unresolved external symbol **PQexec** referenced in function "private: int __cdecl PostGISRasterDataset::SetRasterProperties(char const *)" (?SetRasterProperties@PostGISRasterDataset@@AEAAHPEBD@Z)
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQexec
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQexec
postgisrasterdataset.obj : error LNK2019: unresolved external symbol **PQresultStatus** referenced in function "private: int __cdecl PostGISRasterDataset::SetRasterProperties(char const *)" (?SetRasterProperties@PostGISRasterDataset@@AEAAHPEBD@Z)
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQresultStatus
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQresultStatus
postgisrasterdataset.obj : error LNK2019: unresolved external symbol **PQntuples** referenced in function "private: int __cdecl PostGISRasterDataset::ConstructOneDatasetFromTiles(struct pg_result *)" (?ConstructOneDatasetFromTiles@PostGISRasterDataset@@AEAAHPEAUpg_result@@@Z)
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQntuples
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQntuples
postgisrasterdataset.obj : error LNK2019: unresolved external symbol **PQgetvalue** referenced in function "private: int __cdecl PostGISRasterDataset::ConstructOneDatasetFromTiles(struct pg_result *)" (?ConstructOneDatasetFromTiles@PostGISRasterDataset@@AEAAHPEAUpg_result@@@Z)
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQgetvalue
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQgetvalue
postgisrasterdataset.obj : error LNK2019: unresolved external symbol **PQclear** referenced in function "private: int __cdecl PostGISRasterDataset::SetRasterProperties(char const *)" (?SetRasterProperties@PostGISRasterDataset@@AEAAHPEBD@Z)
postgisrasterrasterband.obj : error LNK2001: unresolved external symbol PQclear
postgisrastertilerasterband.obj : error LNK2001: unresolved external symbol PQclear
postgisrasterdriver.obj : error LNK2019: unresolved external symbol **PQconnectdb** referenced in function "public: struct pg_conn * __cdecl PostGISRasterDriver::GetConnection(char const *,char const *,char const *,char const *,char const *)" (?GetConnection@PostGISRasterDriver@@QEAAPEAUpg_conn@@PEBD0000@Z)
postgisrasterdriver.obj : error LNK2019: unresolved external symbol **PQfinish** referenced in function "public: virtual __cdecl PostGISRasterDriver::~PostGISRasterDriver(void)" (??1PostGISRasterDriver@@UEAA@XZ)
postgisrasterdriver.obj : error LNK2019: unresolved external symbol **PQstatus** referenced in function "public: struct pg_conn * __cdecl PostGISRasterDriver::GetConnection(char const *,char const *,char const *,char const *,char const *)" (?GetConnection@PostGISRasterDriver@@QEAAPEAUpg_conn@@PEBD0000@Z)
gdal204.dll : fatal error LNK1120: 9 unresolved externals

**Comment by JuttaAnders** [comment]
Have a look at the file "ports/gdal/portfile.cmake" line 208:
`list(APPEND NMAKE_OPTIONS_REL PG_LIB=${PGSQL_LIBRARY_DBG})`
and replace it with:
`list(APPEND NMAKE_OPTIONS_DBG PG_LIB=${PGSQL_LIBRARY_DBG})`

**Comment by rkcarroll** [comment]
Thanks! per below, both release and debug built, looks like there's still an install issue:
-- Building x64-windows-rel
-- Building x64-windows-rel done
-- Building x64-windows-dbg
-- Building x64-windows-dbg done
-- Packaging x64-windows
CMake Error at ports/gdal/portfile.cmake:274 (file):
  file must be called with at least two arguments.
Call Stack (most recent call first):
  scripts/ports.cmake:71 (include)

**Comment by Asozial** [comment]
The problem lies in `vcpkg/ports/gdal/portfile.cmake line 274` : `file(REMOVE_RECURSE ${GDAL_TOOLS})`
Maybe ${GDAL_TOOLS} is empty and that is causing the error!?

**Comment by JuttaAnders** [comment]
The problem is the patch file for the static build.
It is ment only for static build.
Pls modify the port file as follow:

```
foreach(BUILD_TYPE IN LISTS BUILD_TYPES)
    file(REMOVE_RECURSE ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-${BUILD_TYPE})
    vcpkg_extract_source_archive(${ARCHIVE} ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-${BUILD_TYPE})
    vcpkg_apply_patches(
        SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-${BUILD_TYPE}/gdal-${GDAL_VERSION_STR}
        PATCHES
        ${CMAKE_CURRENT_LIST_DIR}/0001-Fix-debug-crt-flags.patch
    )
    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        vcpkg_apply_patches(
            SOURCE_PATH ${CURRENT_BUILDTREES_DIR}/src-${TARGET_TRIPLET}-${BUILD_TYPE}/gdal-${GDAL_VERSION_STR}
            PATCHES
                 ${CMAKE_CURRENT_LIST_DIR}/0002-Fix-static-build.patch
        )
    endif()
endforeach()
```

**Comment by LarryIII** [comment]
@JuttaAnders I noticed that caused this PR https://github.com/Microsoft/vcpkg/pull/5839, could you please submit a new PR to fix current issues? Thank you!

**Comment by wangli28** [comment]
Hi @cdcseacave, this issue has been fixed in #5940. You can update vcpkg and rebuild.
Thanks.
